bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -0,0 +1,901 @@
+            setPasswordFromCookie();
+        } catch (ExecutionException e) {
+            logger.warn("ExecutionException: {}", e.getMessage(), e);
+        } catch (InterruptedException e) {
+            logger.warn("InterruptedException: {}", e.getMessage(), e);
+        } catch (TimeoutException e) {
+            logger.warn("TimeoutException: {}", e.getMessage(), e);
+        }
+
+        url = AUTH_LOGIN;
+        return postVerisureAPI(url, "empty");
+    }
+
+    private boolean getInstallations() {
+        logger.debug("Attempting to get all installations");
+
+        int httpResultCode = setSessionCookieAuthLogin();
+        if (httpResultCode == HttpStatus.OK_200) {
+            String url = START_GRAPHQL;
+
+            String queryQLAccountInstallations = "[{\"operationName\":\"AccountInstallations\",\"variables\":{\"email\":\""
+                    + userName
+                    + "\"},\"query\":\"query AccountInstallations($email: String!) {\\n  account(email: $email) {\\n    owainstallations {\\n      giid\\n      alias\\n      type\\n      subsidiary\\n      dealerId\\n      __typename\\n    }\\n    __typename\\n  }\\n}\\n\"}]";
+            Class<VerisureInstallations> jsonClass = VerisureInstallations.class;
+            VerisureInstallations installations = postJSONVerisureAPI(url, queryQLAccountInstallations, jsonClass);
+
+            if (installations == null) {
+                logger.debug("Failed to get installations");
+            } else {
+                logger.debug("Installation: {}", installations.toString());
+                if (installations.getData() != null) {
+                    List<Owainstallation> owaInstList = installations.getData().getAccount().getOwainstallations();
+                    Boolean pinCodesMatchInstallations = true;
+                    List<String> pinCodes = null;
+                    if (pinCode != null) {
+                        pinCodes = Arrays.asList(pinCode.split(","));
+                        if (owaInstList != null && owaInstList.size() != pinCodes.size()) {
+                            logger.debug("Number of installations {} does not match number of pin codes configured {}",
+                                    owaInstList.size(), pinCodes.size());
+                            pinCodesMatchInstallations = false;
+                        } else if (owaInstList == null) {
+                            logger.warn("Failed to get Verisure installations");
+                            return false;
+                        }
+                    } else {
+                        logger.debug("No pin-code defined for user {}", userName);
+                    }
+                    for (int i = 0; i < owaInstList.size(); i++) {
+                        VerisureInstallation vInst = new VerisureInstallation();
+                        if (owaInstList.get(i).getAlias() != null && owaInstList.get(i).getGiid() != null) {
```

Comment:
Please refactor `owaInstList.get(i)` into a local variable so that it can be reused.

Reply:
Fixed!